src/instructlab/config/init.py

```diff
@@ -72,14 +77,83 @@ def init(
 ):
     """Initializes environment for InstructLab"""
     ensure_storage_directories_exist()
+    cfg = None
+    # If the caller used `ilab init` then the parent command is `ilab`, so we need to get the
+    # parameter source from the parent command.
+    if ctx.parent.command.name == "ilab":
+        param_source = ctx.parent.get_parameter_source("config_file")
+    # if the caller used `ilab config init` then the parent command is `config`, so we need to get
+    # the parameter source from the parent of the parent command.
+    else:
+        param_source = ctx.parent.parent.get_parameter_source("config_file")
+
+    if param_source == click.core.ParameterSource.ENVIRONMENT:
+        model_path, taxonomy_path, taxonomy_base, cfg = get_params_from_env(ctx.obj)
+    else:
+        model_path, taxonomy_path, cfg = get_params_default(
+            interactive,
+            repository,
+            min_taxonomy,
+            model_path,
+            taxonomy_path,
+        )
+        # get_params_default() return empty values if the user decides to not continue when a config file is found
+        if model_path == "":
+            ctx.exit()
+
+    click.echo(f"Generating `{DEFAULTS.CONFIG_FILE}`...")
+    if train_profile is not None:
+        cfg.train = read_train_profile(train_profile)
+
+    cfg.chat.model = model_path
+    cfg.generate.model = model_path
+    cfg.serve.model_path = model_path
+    cfg.generate.taxonomy_path = taxonomy_path
+    cfg.generate.taxonomy_base = taxonomy_base
+    cfg.evaluate.model = model_path
+    cfg.evaluate.mt_bench_branch.taxonomy_path = taxonomy_path
+    write_config(cfg)
+
+    click.secho(
+        "Initialization completed successfully, you're ready to start using `ilab`. Enjoy!",
+        fg="green",
+    )
+
+
+def get_params_from_env(
+    obj: typing.Optional[typing.Any],
+) -> typing.Tuple[str, str, str, Config]:
+    if obj is None or not hasattr(obj, "config"):
+        raise ValueError("obj must not be None and must have a 'config' attribute")
+
+    return (
+        obj.config.serve.model_path,
+        obj.config.generate.taxonomy_path,
+        obj.config.generate.taxonomy_base,
+        obj.config,
+    )
+
+
+def get_params_default(
+    interactive: bool,
+    repository: str,
+    min_taxonomy: bool,
+    model_path: str,
+    taxonomy_path: pathlib.Path,
+) -> typing.Tuple[str, pathlib.Path, Config]:
+    cfg = get_default_config()
     clone_taxonomy_repo = True
     if interactive:
         if exists(DEFAULTS.CONFIG_FILE):
             overwrite = click.confirm(
                 f"Found {DEFAULTS.CONFIG_FILE} in the current directory, do you still want to continue?"
             )
             if not overwrite:
-                return
+                return (
+                    "",
+                    pathlib.Path("."),
+                    get_default_config(),
```

Comment:
You could return `cfg` since you already have it. That said, I think you could raise an exception and catch it outside as an indication of exit state, instead of reusing return value for indication about this except-ional situation.

Reply:
Addressed, both.